sencha-workspace/SlateTasksTeacher/app/view/StudentsGrid.js

```diff
@@ -2,312 +2,233 @@
-                            { text: 'Foo' },
-                            { text: 'Foole' },
-                            { text: 'Foo' },
-                            { text: 'Fooianson' },
-                            { text: 'Foods' },
-                            { text: 'Fooo' },
-                            { text: 'Foo' },
-                            { text: 'Foo' },
-                            { text: 'Foole' },
-                            { text: 'Foo' },
-                            { text: 'Fooianson' },
-                            { text: 'Foods' },
-                            { text: 'Fooo' },
-                            { text: 'Foo' },
-                            { text: 'Foo' }
-                        ]
+                getDisplayValue: function(studentTask) {
+                    var html = '';
+
+                    if (studentTask.get('TaskStatus') === 'completed') {
+                        html = '<i class="fa fa-lg fa-check-circle-o"></i>';
+                    } else if (studentTask.get('DueDate')) {
+                        html = Ext.Date.format(studentTask.get('DueDate'), 'm/d');
                     }
-                ]
-            },
+
+                    return html;
+                }
+            }
+        ],
+
+        subCellTpl: [
+            '<tpl if="values.records && values.records[0]">',
+                '{[this.getDisplayValue(values.records[0].record)]}',
+            '</tpl>',
             {
-                title: 'Performance Task 2',
-                students: [
-                    { text: 'Foo' },
-                    { text: 'Foole' },
-                    { text: 'Foo' },
-                    { text: 'Fooianson' },
-                    { text: 'Foods' },
-                    { text: 'Fooo' },
-                    { text: 'Foo' },
-                    { text: 'Foo' },
-                    { text: 'Foole' },
-                    { text: 'Foo' },
-                    { text: 'Fooianson' },
-                    { text: 'Foods' },
-                    { text: 'Fooo' },
-                    { text: 'Foo' },
-                    { text: 'Foo' },
-                    { text: 'Foole' },
-                    { text: 'Foo' },
-                    { text: 'Fooianson' },
-                    { text: 'Foods' },
-                    { text: 'Fooo' },
-                    { text: 'Foo' },
-                    { text: 'Foo' }
-                ]
+                getDisplayValue: function(studentTask) {
+                    var html = '';
+
+                    if (studentTask.get('TaskStatus') === 'completed') {
+                        html = '<i class="fa fa-lg fa-check-circle-o"></i>';
+                    } else if (studentTask.get('DueDate')) {
+                        html = Ext.Date.format(studentTask.get('DueDate'), 'm/d');
+                    }
+
+                    return html;
+                }
             }
-        ]
-    },
+        ],
+
+        tpl: [
```

Comment:
@nbey why are all these structurela tpls' being overridden in this implementation? They're not supposed to be customized and overriding them here is dangerous